### PR TITLE
DAH-2379: Delete app on My Applications page

### PR DESCRIPTION
**Pull request by cliu02**
[DAH-2379](https://sfgovdt.jira.com/browse/DAH-2379)

Note: this branch includes DAH-2342. 

[DAH-2379]: https://sfgovdt.jira.com/browse/DAH-2379?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/api/authApiService.ts [hunk 1]
```diff
@@ -16,8 +16,16 @@ export const signIn = async (email: string, password: string): Promise<User> =>
...
-export const getApplications = async (): Promise<Application[]> =>
-  authenticatedGet<Application[]>("/api/v1/account/my-applications").then((res) => res.data)
+export const getApplications = async (): Promise<{ applications: Application[] }> =>
+  authenticatedGet<{ applications: Application[] }>("/api/v1/account/my-applications").then(
+    (res) => res.data
+  )
+
+export const deleteApplication = async (id: string) =>
+  authenticatedDelete(`/api/v1/short-form/application/${id}`).then((res) => {
+    console.log("res.data", res.data)
```
Comment (jimlin-sfgov) [inline app/javascript/api/authApiService.ts:26]: issue: Remove debugging statement

File: app/javascript/components/ApplicationItem.tsx [hunk 2]
```diff
@@ -4,120 +4,208 @@ import {
...
-  lotteryError?: boolean
-  lotteryResultsURL?: string
   submitted: boolean
-  pastDue?: boolean
+  handleDeleteApp?: (id: string) => void
+  listing: RailsListing
+  lotteryResultsURL?: string
+}
+
+export const convertToReadableDate = (dateString: string) => {
```
Comment (jimlin-sfgov) [inline app/javascript/components/ApplicationItem.tsx:36]: suggestion: move exportable function to a helper/utility file.

**Review by tallulahkay — CHANGES_REQUESTED** [review]
The success path here works! But if there's a failure to delete the application (salesforce api returns anything other than 200), you get: 
Screenshot 2024-07-08 at 4 13 01 PM
and then when you refresh the page, you still see the application

In the angular version, failure looks like this: 
Screenshot 2024-07-08 at 4 14 29 PM

Wondering if it's been discussed with Yindi what failure should look like? I think we should aim for parity with how the angular application currently works when there is an api failure

**Review by tallulahkay — APPROVED** [review]
yay!!

**Review by jimlin-sfgov — APPROVED** [review]
@chadbrokaw Looking forward to getting https://github.com/SFDigitalServices/sf-dahlia-web/pull/2207 merged

**Comment by jimlin-sfgov** [comment]
Oops, looks like we let this get through without proper coverage checks? I think we'll need to improve coverage in the related PR: https://github.com/SFDigitalServices/sf-dahlia-web/pull/2207

**Comment by chadbrokaw** [comment]
> @chadbrokaw Looking forward to getting #2207 merged

Me too! Just got updated copy for the my applications page so should be good to go today 🙏

**Pull request merged**